### PR TITLE
Make setup script more robust and idempotent

### PR DESCRIPTION
**Pull request by evanfoster**
While doing some development work for a different feature, I kept running into issues with `make setup`. I'd run it after rebooting my laptop and it would fail, or I'd run it and receive a weird error that I wouldn't be able to reproduce. To address this, I've added more robust error handling and have taken the opportunity to make the script idempotent. You should be able to run `make setup` on a fully running system and have nothing happen except for the API server and client containers roll.

### CODE REVIEW

File: local/setup.sh [hunk 1]
```diff
@@ -1,8 +1,45 @@
 #!/usr/bin/env bash
-
-set -o errexit
```
Comment (evanfoster) [inline local/setup.sh:3]: `set -e` comes with some pretty major drawbacks. I've found that it's generally better to have a `die` function that you call when running an important command. [This wiki](https://mywiki.wooledge.org/BashFAQ/105) does a great job of explaining why `set -e` is problematic. It's very likely that `errexit`, `nounset`, or `pipefail` were responsible for some of the flakiness I was seeing.

**Review by aalexandru — APPROVED** [review]

File: local/setup.sh [hunk 2]
```diff
@@ -11,113 +48,143 @@ RUN_CLIENT="${2:-1}"
...
+        -e SQS_AWS_REGION \
+        -e SQS_ENDPOINT=http://"${CONTAINER_SQS}":9324 \
+        -e SQS_QUEUE_NAME="${SQS_QUEUE_NAME}" \
+        --network "${NETWORK}" \
+        "${IMAGE_CLIENT}":"${TAG}" || die "Failed to create $CONTAINER_CLIENT container."
 fi
 
 echo 'Local stack was set up successfully.'
+
+# vim: ts=4 sw=4 et
```
Comment (radu-catalina) [inline local/setup.sh:190]: what do you think of using https://editorconfig.org/ to store these options?

**Review by radu-catalina — APPROVED** [review]
lgtm

File: local/setup.sh (hunk 2 above)
Comment (evanfoster) [reply local/setup.sh]: :facepalm: I totally missed that! Let me move it there.

**Comment by evanfoster** [comment]
Hmm. I've signed the CLA, but I must've done something wrong.

**Pull request merged**